pandas/tests/test_window.py

```diff
@@ -441,6 +441,20 @@ def test_closed(self):
         with pytest.raises(ValueError):
             df.rolling(window=3, closed='neither')
 
+    def test_empty_df_datetime_rolling_sum(self):
+        # Verifies that datetime rolling window can be applied to empty DataFrame
+        # GH 15819
+        result = DataFrame().rolling('1s').sum()
+        expected = DataFrame()
+        tm.assert_frame_equal(result, expected)
+
+    def test_empty_df_integer_rolling_sum(self):
+        # Verifies that integer rolling window can be applied to empty DataFrame
+        # GH 15819
```

Comment:
can you add the same tests for expanding (not that the expanding will raise for the time-aware, a separate issue, so xfail that one, put a ref to #16425 there).